### PR TITLE
fix/test quant

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

fix skip tests and add test for torchscript

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
Looks great !

File: pytorch_lightning/utilities/imports.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def _compare_version(package: str, op, version) -> bool:
 _TORCH_LOWER_EQUAL_1_4 = _compare_version("torch", operator.le, "1.5.0")
 _TORCH_GREATER_EQUAL_1_6 = _compare_version("torch", operator.ge, "1.6.0")
 _TORCH_GREATER_EQUAL_1_7 = _compare_version("torch", operator.ge, "1.7.0")
-_TORCH_QUANTIZE_AVAILABLE = _module_available('torch.ops.quantized')
+_TORCH_QUANTIZE_AVAILABLE = hasattr(torch.ops, "quantized")
```
Comment (carmocca) [inline pytorch_lightning/utilities/imports.py:55]: Torch 1.4.0 already has quantization. Should we drop this check instead?

**Review by SkafteNicki — APPROVED** [review]
LGTM :]

File: pytorch_lightning/utilities/imports.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/utilities/imports.py]: I am checking what is the issue and turned out that mac for some reason has 
```
torch.backends.quantized.supported_engines <- ['none']
```

**Review by SeanNaren — APPROVED** [review]

File: pytorch_lightning/callbacks/quantization.py [hunk 2]
```diff
@@ -128,8 +128,11 @@ def custom_trigger_last(trainer):
             input_compatible: preserve quant/dequant layers. This allows to feat any input as to the original model,
                 but break compatibility to torchscript
         """
-        if not isinstance(qconfig, (str, QConfig)):
-            raise MisconfigurationException(f"Unsupported qconfig: f{qconfig}.")
+        _valid_gconf_str = isinstance(qconfig, str) and qconfig in torch.backends.quantized.supported_engines
+        if not isinstance(qconfig, QConfig) and not _valid_gconf_str:
```
Comment (carmocca) [inline pytorch_lightning/callbacks/quantization.py:132]:
```suggestion
        _valid_qconf_str = isinstance(qconfig, str) and qconfig in torch.backends.quantized.supported_engines
        if not isinstance(qconfig, QConfig) and not _valid_qconf_str:
```

File: pytorch_lightning/callbacks/quantization.py [hunk 3]
```diff
@@ -128,8 +128,11 @@ def custom_trigger_last(trainer):
...
+            raise MisconfigurationException(
+                f"Unsupported qconfig: f{qconfig}.\n Try one of defaults: {torch.backends.quantized.supported_engines}"
```
Comment (carmocca) [inline pytorch_lightning/callbacks/quantization.py:134]: No need for the space with the line break
```suggestion
                f"Unsupported qconfig: f{qconfig}.\nTry one of defaults: {torch.backends.quantized.supported_engines}"
```

File: pytorch_lightning/utilities/imports.py (hunk 1 above)
Comment (carmocca) [reply pytorch_lightning/utilities/imports.py]: that's weird 🤯

**Review by carmocca — APPROVED** [review]

**Review by ananyahjha93 — APPROVED** [review]

File: pytorch_lightning/callbacks/quantization.py (hunk 3 above)
Comment (Borda) [reply pytorch_lightning/callbacks/quantization.py]: true, but it is easier to read since it is on the same line :]

**Review by s-rog — APPROVED** [review]

**Pull request merged**